iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/base_relative_pointer.hpp

```diff
@@ -170,12 +191,15 @@ class BaseRelativePointer
     /// @return the pointer for stored id and offset
     ptr_t computeRawPtr() const noexcept;
 
-    static constexpr id_t NULL_POINTER_ID = std::numeric_limits<id_t>::max();
```

Comment:
@elBoberido @MatthiasKillat 
Unfortunately, due to the default d'tors in `NewType`, it can't be `constexpr`. We could however remove the default d'tors in `NewType` and then `id_t` could stay `constexpr`.

Reply:
@mossmaurice the problem is not the dtor. It must be something else. This compiles and runs
```cpp
#include <cstdint>
#include <iostream>

struct Foo {
    constexpr Foo(uint64_t size) : size(size) {}

    uint64_t size {0};
protected:
    ~Foo() = default;
};

struct Bar : public Foo {
    constexpr Bar(uint64_t size) : Foo(size) {}
};

template <uint64_t N>
uint64_t answer() {
    return N;
}

int main() {
    static constexpr Bar bar{42};

    std::cout << answer<bar.size>() << std::endl;

    return 0;
}
```

Does the `NewType` have `constexpr` ctors?

Reply:
@elBoberido I tried to make `NewType` `constexpr` by adding it for all the methods. However, I couldn't make it compile :( We should try that on a separate branch maybe.

Reply:
btw, the `id_t` is in the `rp` namespace, right?